### PR TITLE
Avoid JavaScript and Python in Enso native launcher

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

To use Python or JavaScript interop one has to specify `--jvm` flag.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
- [x] Unit tests shall continue to pass - already fixed in #10975

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Failure: No language for [regex](https://github.com/enso-org/enso/actions/runs/11250063690/job/31278178142?pr=11279#step:7:2272) found in tests. Addressed in acf44d0

**Comment by JaroslavTulach** [comment]
We are able to execute the `Dictionary_Spec` tests in **3s**!
```
$ time ./built-distribution/enso-engine-*/enso-*/bin/enso --run test/Base_Tests/src/Data/Dictionary_Spec.enso 
[PENDING] JavaScript: Common polyglot Dictionary operations
    Reason: Can't run JavaScript tests, language `js` is not installed.
[PENDING] Python: Common polyglot Dictionary operations
    Reason: Can't run Python tests, Python is not installed.

90 tests succeeded.
0 tests failed.
16 tests skipped.
2 groups skipped.


real    0m3,154s
user    0m8,067s
sys     0m0,545s
```
Sixteen tests is skipped because of intentionally missing Python and JavaScript. Adding `--jvm` option to the command runs all tests, but (also due to initialization of Python and JavaScript) takes 20s.

File: distribution/engine/THIRD-PARTY/NOTICE [hunk 1]
```diff
@@ -431,26 +416,11 @@ The license file can be found at `licenses/MIT`.
...
 
 
-'python-language', licensed under the MIT License, is distributed with the engine.
-The license file can be found at `licenses/MIT`.
-Copyright notices related to this dependency can be found in the directory `org.graalvm.python.python-language-24.0.0`.
-
-
-'python-resources', licensed under the MIT License, is distributed with the engine.
-The license file can be found at `licenses/MIT`.
-Copyright notices related to this dependency can be found in the directory `org.graalvm.python.python-resources-24.0.0`.
```
Comment (radeusgd) [inline distribution/engine/THIRD-PARTY/NOTICE]: Are we no longer distributing the Python language with Enso? I thought that with `--jvm` option it is still somehow picked up, but how is it picked up if it's not present in the distribution?

Comment (radeusgd) [reply distribution/engine/THIRD-PARTY/NOTICE]: (I probably don't understand something, so I'd appreciate an explanation to be able to understand what has changed in terms of Python availability/distribution with our artifacts)

Comment (JaroslavTulach) [reply distribution/engine/THIRD-PARTY/NOTICE]: No, we do distribute `python-resources` and co. Those JARs still get copied into `component` directory. Just the legal tool isn't able to figure it out. Who knows where exactly it finds the JARs to analyze?

Comment (radeusgd) [reply distribution/engine/THIRD-PARTY/NOTICE]: All sbt projects that are distributed are listed in `GatherLicenses.distributions := ` setting.

For engine, that is:
```
    Distribution.sbtProjects(
      runtime,
      `engine-runner`,
      `language-server`
    )
```

Should we add something more?

Comment (JaroslavTulach) [reply distribution/engine/THIRD-PARTY/NOTICE]: Thanks a handy advice! Fixed in d8a42b3

File: build.sbt [hunk 2]
```diff
@@ -2743,6 +2744,47 @@ lazy val runtime = (project in file("engine/runtime"))
...
+      (`cli` / Compile / exportedModule).value,
+      (`editions` / Compile / exportedModule).value,
+      (`edition-updater` / Compile / exportedModule).value,
+      (`syntax-rust-definition` / Compile / exportedModule).value,
+      (`version-output` / Compile / exportedModule).value,
+      (`interpreter-dsl` / Compile / exportedModule).value,
+      (`persistance` / Compile / exportedModule).value,
+      (`text-buffer` / Compile / exportedModule).value,
+      (`scala-libs-wrapper` / Compile / exportedModule).value,
+      (`fansi-wrapper` / Compile / exportedModule).value
```
Comment (radeusgd) [inline build.sbt]: How did you decide which of the sub-projects should be included on that list and which not?

File: build.sbt [hunk 3]
```diff
@@ -2743,6 +2744,47 @@ lazy val runtime = (project in file("engine/runtime"))
...
+    annotationProcSetting,
+    truffleDslSuppressWarnsSetting,
+    version := ensoVersion,
+    commands += WithDebugCommand.withDebug,
+    inConfig(Compile)(truffleRunOptionsSettings),
+    // Add all GraalVM packages with Runtime scope - we don't need them for compilation,
+    // just provide them at runtime (in module-path).
+    libraryDependencies ++= {
+      GraalVM.modules ++ GraalVM.langsPkgs
+    },
```
Comment (radeusgd) [inline build.sbt]: 👍

**Review by radeusgd — APPROVED** [review]
Looks good that we now don't have any license changes 👍 

I'd still like to understand what was the reasoning behind `internalModuleDependencies` in `runtime-and-langs`. If we ever need to update it - how do I know what I need to include there and what not? Could you please add some explanation?

File: build.sbt [hunk 4]
```diff
@@ -2743,6 +2744,47 @@ lazy val runtime = (project in file("engine/runtime"))
   .dependsOn(`connected-lock-manager`)
   .dependsOn(testkit % Test)
 
+lazy val `runtime-and-langs` = (project in file("engine/runtime-and-langs"))
```
Comment (Akirathan) [inline build.sbt:2747]: It is better to provide `runtime-and-langs` as an aggregate project, similar to `enso`. It is afterall, merely an aggregation of multiple projects. You don't need any additional settings for that, you just need to specify the `aggregate`.

**Review by Akirathan — CHANGES_REQUESTED** [review]
Overall removal of dependencies seems good - we don't need python neither javascript for runtime testing, and for other stuff. But please simplify the definition of `runtime-with-langs` project definition - it can be implemented simply as something like:
```
lazy val `runtime-with-langs` = (project in fille("runtime-with-langs"))
  .aggregate(
    <list-of-all-the-projects>
  )
```

I would like to avoid adding something that resembles `runtime-fat-jar` as much as possible.

Another motivation for using aggregate is that it will not create as many sbt settings, thus, the load tim and build time should be a little bit better.

File: build.sbt (hunk 2 above)
Comment (JaroslavTulach) [reply build.sbt]: I copied that from runtime

File: build.sbt (hunk 4 above)
Comment (JaroslavTulach) [reply build.sbt:2747]: Simplified in 147492f- is that enough?

File: build.sbt (hunk 2 above)
Comment (JaroslavTulach) [reply build.sbt]: Simplified in 147492f

File: build.sbt (hunk 4 above)
Comment (Akirathan) [reply build.sbt:2747]: Thaty is a lot better, thank you.

**Review by Akirathan — APPROVED** [review]

**Pull request merged**